be/src/storage/persistent_index.cpp

```diff
@@ -758,19 +782,72 @@ Status PersistentIndex::abort() {
 
 Status PersistentIndex::commit(PersistentIndexMetaPB* index_meta) {
     // TODO: l0 may be need to flush
-    MutableIndexMetaPB* l0_meta = index_meta->mutable_l0_meta();
-    IndexWalMetaPB* wal_pb = l0_meta->add_wals();
-    _version.to_pb(wal_pb->mutable_version());
+    if (_dump_snapshot) {
+        // if _map size is small enough to dump directly, rewrite snapshot
+        std::string file_name = _get_l0_index_file_name(_path, _version);
+        // be maybe crash after create index file during last commit
+        // so we delete expired index file first to make sure no garbage left
+        Env::Default()->delete_file(file_name);
+        fs::BlockManager* block_mgr = fs::fs_util::block_manager();
+        std::unique_ptr<fs::WritableBlock> wblock;
+        fs::CreateBlockOptions wblock_opts({file_name});
+        RETURN_IF_ERROR(block_mgr->create_block(wblock_opts, &wblock));
+        DeferOp close_block([&wblock] {
+            if (wblock) {
+                wblock->close();
+            }
+        });
```

Comment:
> Is this neccesary?

The above code is not need because new index file will be created during `open` if not exist, change in https://github.com/StarRocks/starrocks/pull/3654/commits/a3d4d0f762f160aac3e481a90df47ab5ee26c585